src/material-experimental/mdc-slider/testing/slider-harness.ts

```diff
@@ -6,9 +6,63 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {ComponentHarness} from '@angular/cdk/testing';
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {coerceNumberProperty} from '@angular/cdk/coercion';
+import {SliderHarnessFilters, ThumbPosition} from './slider-harness-filters';
+import {MatSliderThumbHarness} from './slider-thumb-harness';
 
 /** Harness for interacting with a MDC mat-slider in tests. */
 export class MatSliderHarness extends ComponentHarness {
-  // TODO(wagnermaciel): Implement this in a separate PR
+  static hostSelector = '.mat-mdc-slider';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatSliderHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which input instances are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: SliderHarnessFilters = {}): HarnessPredicate<MatSliderHarness> {
+    return new HarnessPredicate(MatSliderHarness, options)
+      .addOption('isRange', options.isRange, async (harness, value) => {
+        return (await harness.isRange()) === value;
+      });
+  }
+
+  /** Gets the start/primary thumb of the slider. */
+  async getStartThumb(): Promise<MatSliderThumbHarness> {
+    return this.locatorFor(MatSliderThumbHarness.with({position: ThumbPosition.START}))();
+  }
+
+  /** Gets the end thumb of the slider. Will throw an error for a non-range slider. */
+  async getEndThumb(): Promise<MatSliderThumbHarness> {
```

Comment:
We could follow MDC's implementation ([link](https://github.com/material-components/material-components-web/tree/master/packages/mdc-slider#mdcslider-methods)) and use `getThumb` and `getThumbStart` similar to their `getValue` and `getValueStart`. I'm not saying we _should_ - I actually like the way things are here - but I figured I should leave this suggestion